apps/dashboard/src/components/side-navigation/free-trial-card.tsx

```diff
@@ -0,0 +1,74 @@
+import { useBillingSubscription } from '@/hooks';
+import { LogoCircle } from '../icons';
+import { RiArrowRightDoubleLine, RiInformationFill } from 'react-icons/ri';
+import { Progress } from '../primitives/progress';
+import { Button } from '../primitives/button';
+import { Tooltip, TooltipContent, TooltipProvider, TooltipTrigger, TooltipArrow } from '../primitives/tooltip';
+import { LEGACY_ROUTES } from '@/utils/routes';
+
+const transition = 'transition-all duration-300 ease-out';
+
+const pluralizeDaysLeft = (numberOfDays: number) => {
+  return `${numberOfDays} day${numberOfDays > 1 ? 's' : ''}`;
+};
+
+export const FreeTrialCard = () => {
+  const { subscription, isLoading } = useBillingSubscription();
+  const { trial, hasPaymentMethod } = subscription;
+
+  if (isLoading || !trial.isActive || hasPaymentMethod) {
+    return null;
+  }
```

Comment:
@ChmaraX, is this logic correct?

Reply:
You can remove the `hasPaymentMethod` - currently its there to support the legacy billing, but when we release new billing (which will next week) we don't use `hasPaymentMethod` as a check for trial anymore.

FYI: in new billing, you are able to only buy right away, so there is no state where you would be in trial and have payment method saved